framework/core/js/src/common/Translator.tsx

```diff
@@ -88,4 +89,16 @@ export default class Translator {
 
     return id;
   }
+
+  /**
+   * Formats the time.
+   * 
+   * The format of the time will be chosen by the following order:
+   * - The format ID defined in current locale.
+   * - The provided fallback format.
+   * - DayJS default format.
+   */
+  format(time: Dayjs, id?: string, fallback?: string): string {
+    return time.format(id && (this.translations[id] ?? fallback));
+  }
```

Comment:
Well, I thought about this when adding the `format` method.

Since the extensions can extend/override `format` method directly, so they can just return the value without calling the original one to add custom fields. However, a dedicated method that returns item list seems to be easier to add a new format.

Which way do you prefer?

Reply:
Yea, it's true they can use extend/override helpers. Good point.

I think we can still use an item list but in a cleaner way that overriding the method, and also cleaner than the additional method i proposed. How about a propery item list on the class itself. with a app.translator.addDateTimeFormat(format, () => string)

it's cleaner than having to do override(app.translator.format, ...)

wdyt?

Reply:
Is the `format` argument in `addDateTimeFormat` the `id` argument in `format`? If so, this sure can be a quick way to add a format without extend/override. And extensions can still do so if they want more control over this.

I'll add it if you think it's good.

Reply:
@YUCLing yup I think the format should be the same, even if it's a locale key. I don't personally mind that and keeps things simple. I'd say let's go with that 👍🏼